### PR TITLE
Enable lint for Auth

### PR DESCRIPTION
**Pull request by renkelvin**

### CODE REVIEW

**Comment by ryanwilson** [comment]
Looked at the public API surface to start - nothing affected outside of the few small comments.

File: scripts/style.sh [hunk 1]
```diff
@@ -153,8 +153,11 @@ s%^./%%
 # Sources pulled in by travis bundler, with and without a leading slash
 \%^/?vendor/bundle/% d
 
-# Sources within the tree that are not subject to formatting
-\%^(Example|Firebase)/(Auth|AuthSamples)/% d
+# Messaging is not subject to formatting
+\%^(Example|Firebase)/(Messaging)/% d
+
+# Auth Sample is not subject to formatting
+\%^(Example/Auth/Sample)/% d
```
Comment (renkelvin) [inline scripts/style.sh]: All other changes are made by style.sh

**Comment by ryanwilson** [comment]
Sorry - these were sitting in Draft for the last 3 days for some reason.

File: Firebase/Auth/Source/Public/FIRUser.h [hunk 2]
```diff
@@ -291,8 +292,7 @@ DEPRECATED_MSG_ATTRIBUTE( "Please use reauthenticateWithCredential:completion: f
 - (void)reauthenticateWithProvider:(id<FIRFederatedAuthProvider>)provider
                         UIDelegate:(nullable id<FIRAuthUIDelegate>)UIDelegate
                         completion:(nullable FIRAuthDataResultCallback)completion
-                        NS_SWIFT_NAME(reauthenticate(with:uiDelegate:completion:))
-                        API_AVAILABLE(ios(8.0));
+    NS_SWIFT_NAME(reauthenticate(with:uiDelegate:completion:))API_AVAILABLE(ios(8.0));
```
[the remark this answers is not in the document]
Comment (renkelvin) [reply Firebase/Auth/Source/Public/FIRUser.h]: Honestly it looks like a clang-format bug to me. I'll go with option 1.

File: scripts/style.sh [hunk 3]
```diff
@@ -153,8 +153,11 @@ s%^./%%
 # Sources pulled in by travis bundler, with and without a leading slash
 \%^/?vendor/bundle/% d
 
-# Sources within the tree that are not subject to formatting
-\%^(Example|Firebase)/(Auth|AuthSamples)/% d
+# Messaging is not subject to formatting
+\%^(Example|Firebase)/(Messaging)/% d
```
[the remark this answers is not in the document]
Comment (renkelvin) [reply scripts/style.sh]: Seems it's because of a problem with the rebasing. Thanks for catching that!

**Comment by renkelvin** [comment]
> Sorry - these were sitting in Draft for the last 3 days for some reason.

No worries. I was waiting for the travis then didn't request reviews.

**Pull request merged**